src/lib.rs

```diff
@@ -282,18 +282,30 @@ fn parse_unary_expression(input: ParseInput) -> ParseResult<(UnaryOperator, Expr
 /// Parses a binary operator. E.g. `|`, `&`, etc.
 fn parse_binary_operator(input: ParseInput) -> ParseResult<BinaryOperator> {
     let op = alt((
-        // These two must be first, else | and & are matched
+        // These two must match before | and &
```

Comment:
OOC why? Wouldn't a parse matching this as bitwise or, bitwise or fail and backtrack here? Even if so, this'd be faster, so I'm just curious.

Reply:
Hmm, interesting question. I think `alt` always returns the first successful match. I think `alt`'s backtracking only works 'within' alt. Not between invocations of a given alt.  So if the caller of this alt called it a second time (because it's backtracking), alt would simply match the same thing again.